src/streamlink/plugins/ceskatelevize.py

```diff
@@ -5,131 +5,177 @@
 $region Czechia
 """
 
-import logging
+import json
 import re
+from urllib.parse import urlparse
 
-from streamlink.plugin import Plugin, PluginError, pluginmatcher
+from streamlink.plugin import Plugin, pluginmatcher
 from streamlink.plugin.api import validate
 from streamlink.stream.dash import DASHStream
 
-log = logging.getLogger(__name__)
 
-
-@pluginmatcher(re.compile(
-    r"https://(?:www\.)?ceskatelevize\.cz/zive/\w+"
-))
+@pluginmatcher(re.compile(r"https?://ct24\.ceskatelevize\.cz/"))
+@pluginmatcher(re.compile(r"https?://decko\.ceskatelevize\.cz/"))
+@pluginmatcher(re.compile(r"https?://sport\.ceskatelevize\.cz/(?:zive-vysilani)?#live"))
```

Comment:
I see what you're saying - the page data is still the same with or without the hash.  I think it will still need to be included to match if it's used and to avoid other unintended matches?  For that reason, I've settled on:
```
@pluginmatcher(re.compile(r"https?://sport\.ceskatelevize\.cz/(?:zive-vysilani)?(?:#live$)?$"))
```
Does that seem reasonable?

Reply:
> I think it will still need to be included to match if it's used and to avoid other unintended matches?

No, because of the TLD which contains "sport."

Reply:
I think I'm missing something here.  Unlike `get_ct24()`, `get_sport()` parses the page content of `self.url`, so don't I want to limit the path matches?

Reply:
I don't understand what you're saying. What does this have to do with the hash? Hashes are for client-side DOM stuff and not for HTTP requests, so the `#live` hash should not be included, because it won't affect the response content which you're parsing. And other matchers can't be matched, because the TLD of every matcher is distinctive.

It's just a detail whether the `#live` hash included or not, but including hashes in general is just unnecessary.

Reply:
What regex would you be using here?  Perhaps that would help to clear things up.

Reply:
One without the `#live` hash

Reply:
OK, so I think you want:
```
@pluginmatcher(re.compile(r"https?://sport\.ceskatelevize\.cz/(?:zive-vysilani)?$"))
```
While I recognise what you're saying about `#live` being redundant for the purposes of the page data returned:
```
$ streamlink 'https://sport.ceskatelevize.cz/#live'
error: No plugin can handle URL: https://sport.ceskatelevize.cz/#live
```
My argument here would be that users expect to be able to copy/paste the URL from the browser address bar as is, regardless of whether or not bits of it are technically unnecessary.

I don't want to drag this out any further, so I can put the above regex in if you feel that's what it should be.

Reply:
> users expect to be able to copy/paste the URL from the browser

Yes, absolutely, but why the `$` anchor?

As said, it's just a detail for being able to input the URL without the hash. If this is causing confusion or any issues, just include the hash in the regex.